src/components/editor/markdown-renderer/replace-components/highlighted-fence/highlighted-fence.tsx

```diff
@@ -1,16 +1,18 @@
 import { DomElement } from 'domhandler'
-import React, { ReactElement } from 'react'
+import React from 'react'
 import { HighlightedCode } from '../../../../common/highlighted-code/highlighted-code'
+import { ComponentReplacer, SubNodeConverter } from '../ComponentReplacer'
 
-const getElementReplacement = (codeNode: DomElement, index: number, counterMap: Map<string, number>): (ReactElement | undefined) => {
-  if (codeNode.name !== 'code' || !codeNode.attribs || !codeNode.attribs['data-highlight-language'] || !codeNode.children || !codeNode.children[0]) {
-    return
-  }
+export class HighlightedCodeReplacer implements ComponentReplacer {
```

Comment:
this file should then be called `highlighted-fence-replacer.tsx`?

Reply:
renamed